pybop/problems/fitting_problem.py

```diff
@@ -20,51 +23,44 @@ class FittingProblem(BaseProblem):
         Dataset object containing the data to fit the model to.
     signal : str, optional
         The variable used for fitting (default: "Voltage [V]").
-    additional_variables : List[str], optional
+    additional_variables : list[str], optional
         Additional variables to observe and store in the solution (default additions are: ["Time [s]"]).
-    init_soc : float, optional
-        Initial state of charge (default: None).
+    init_ocv : float, optional
+        Initial open-circuit voltage (default: None).
     """
 
     def __init__(
         self,
-        model,
-        parameters,
-        dataset,
-        check_model=True,
-        signal=None,
-        additional_variables=None,
-        init_soc=None,
+        model: BaseModel,
+        parameters: Parameters,
+        dataset: Dataset,
+        check_model: bool = True,
+        signal: Optional[list[str]] = None,
+        additional_variables: Optional[list[str]] = None,
+        init_ocv=None,
```

Comment:
When would this functionality be useful?

Reply:
One such area would be fitting data that wasn't stationary at the first measurement. You could then provide an SOC instead of a OCV and proceed with fitting. I'm keen to not remove functionality, but guide users instead.

Reply:
Until we can restrict this functionality to cases where the parameters do not affect the OCV, I think we should guide users towards well-defined problems.

Reply:
Note also that such problems may already be solved, by setting the initial state of the model before passing it to the problem, so no functionality has been removed.

Reply:
I stand by my comments that we should provide better guidance and that any known initial state should be passed to the model, not the problem, but I have changed the code to take any `initial_state` rather than an `init_soc`/`init_ocv`.

Reply:
The functionality mentioned above could and should still be achieved by passing the initial SOC to the model.